### PR TITLE
Add data fetching

### PR DESCRIPTION
**Pull request by VolodymyrButs**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -36,7 +38,7 @@ function App() {
     return (
         <Router>
             <div className="App">
-                {!isLoaded ? (
+                {isLoaded ? (
```
Comment (dkupchyk) [inline src/App.js]: Dont understand why removed ! symbol, now the loader is not working.

File: src/shared/slider/slider.css [hunk 2]
```diff
@@ -20,7 +21,7 @@
     outline: 0;
     transition: all 0.5s;
     border-radius: 35px;
-    z-index: 1000;
+    z-index: 1;
     border: 0;
     background: rgba(0, 0, 0, 0.5);
     min-width: 35px;
```
Comment (dkupchyk) [inline src/shared/slider/slider.css]: Suggest changing to _min-width: 25px;_ and _min-height: 25px;_ so it won`t look so massive.

File: src/shared/slider/slider.css [hunk 3]
```diff
@@ -36,7 +37,7 @@
     color: #fff;
     display: block;
     text-align: center;
-    z-index: 2;
+    z-index: 1;
     position: relative;
 }
 .react-multiple-carousel__arrow--left {
```
Comment (dkupchyk) [inline src/shared/slider/slider.css:43]: Change to:

.react-multiple-carousel__arrow--left {
    top: 35%;
    left: 15px;
}
.react-multiple-carousel__arrow--left::before {
    content: '\1438';
}
.react-multiple-carousel__arrow--right {
    top: 35%;
    right: 15px;
}
.react-multiple-carousel__arrow--right::before {
    content: '\1433';
}

File: src/redux/actions/data.js [hunk 4]
```diff
@@ -16,21 +16,12 @@ export const setWidth = width => ({
...
-        dispatch(setData(response))
-        dispatch(setLoaded(true))
-    }, 2000)
-
-    // TODO
-    /* getDataFromSeverFunction().then(({ data }) => {
-        // TODO: create an object with 4 arrays from the response
-        dispatch(setData(data))
-    }) */
+// dispatch(setLoaded(false))
```
Comment (dkupchyk) [inline src/redux/actions/data.js]: Remove unnecessary code.

File: src/redux/actions/data.js [hunk 5]
```diff
@@ -16,21 +16,12 @@ export const setWidth = width => ({
...
+export const fetchData = token => dispatch => {
+    getDataFromSeverFunction(token)
+        .then(data => data.json())
+        .then(body => {
+            dispatch(setData(body))
+        })
 }
+// dispatch(setLoaded(true))
```
Comment (dkupchyk) [inline src/redux/actions/data.js]: Remove unnecessary code.

File: src/components/AllGames/GameCardSlider/GameCardSlider.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
```
Comment (dkupchyk) [inline src/components/AllGames/GameCardSlider/GameCardSlider.js:1]: Looks a bit strange, GameCardSlider is identical to usual GameCard, suggest using only GameCard.

File: src/components/AllGames/AllGames.jsx [hunk 7]
```diff
@@ -1,50 +1,36 @@
...
+                    <GamesCardSlider
+                        key={item.id}
+                        id={item.id}
+                        title={item.displayName}
+                        image={item.url}
+                    />
+                ))}
+            </Slider>
+            <div className={styles['tags']}>
+                <GamesContainer title={'Все игры'} games={allGames} />
```
Comment (dkupchyk) [inline src/components/AllGames/AllGames.jsx]: Should display all containers for all tags, not only the "All games" tag, add maping throug all tags

File: src/components/AllGames/AllGames.jsx [hunk 8]
```diff
@@ -1,50 +1,36 @@
...
+    const { allGames } = useSelector(allGamesSelector)
+
+    const topGames = allGames.filter(game => game.tags.includes('top'))
+    console.log(allGames, topGames)
+
+    return (
+        <div className={styles['container']}>
+            <Slider>
+                {topGames.map(item => (
+                    <GamesCardSlider
```
Comment (dkupchyk) [inline src/components/AllGames/AllGames.jsx]: Slider doesn`t work if resizing a desktop.

File: src/components/Sidebar/Sidebar.module.css [hunk 9]
```diff
@@ -212,8 +215,8 @@ hr {
 
         overflow-y: unset;
```
Comment (dkupchyk) [inline src/components/Sidebar/Sidebar.module.css]: Remove  overflow-y: unset; and then apply filters button will display correctly.

File: src/components/fetchs/fetchs.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
+import moment from 'moment'
```
Comment (dkupchyk) [inline src/components/fetchs/fetchs.js]: move fetchs to shared folder

File: src/redux/actions/data.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import { response } from '../response'
```
Comment (dkupchyk) [inline src/redux/actions/data.js:1]: delete response.js file in redux folder

File: src/components/AllGames/GameCardSlider/GameCardSlider.js (hunk 6 above)
Comment (VolodymyrButs) [reply src/components/AllGames/GameCardSlider/GameCardSlider.js:1]: These cards different in styles

File: src/components/GamePage/GamePage.jsx [hunk 12]
```diff
@@ -8,15 +8,20 @@ import Header from '../Header/Header'
...
     return (
         <div>
-            <Header isMainPage={false} />
+            <Header isMainPage={false} title={game.displayName} />
             <div className={styles['frame-container']}>
-                <iframe id={id} src={game.url} />
+                {game ? (
+                    <iframe id={id} src={game.url || 'https://wiki.com'} />
+                ) : (
+                    () => history.push('/game/original_dice_game')
```
Comment (STDio96) [inline src/components/GamePage/GamePage.jsx:23]: shouldn't we get this ID from the server?

File: src/components/Sidebar/FilterButton/FilterButton.jsx [hunk 13]
```diff
@@ -50,18 +53,14 @@ function FilterButton({
...
                     onClick={onFilterButtonClick}
                 >
-                    <img src="./icons/settings.svg" alt="filters-icon" />
+                    <Settings />
                     <span>Фильтры</span>
                 </button>
             ) : (
                 <div className={styles['filters-opened']}>
                     <div>
                         <span>Фильтры</span>
```
Comment (STDio96) [inline src/components/Sidebar/FilterButton/FilterButton.jsx:62]: can move that to `shared/messages.js`

File: src/components/Sidebar/FilterButton/FilterButton.jsx [hunk 14]
```diff
@@ -50,18 +53,14 @@ function FilterButton({
                     className={styles['filters-closed']}
                     onClick={onFilterButtonClick}
                 >
-                    <img src="./icons/settings.svg" alt="filters-icon" />
+                    <Settings />
                     <span>Фильтры</span>
```
Comment (STDio96) [inline src/components/Sidebar/FilterButton/FilterButton.jsx:57]: can move that to `shared/messages.js`

File: src/components/Sidebar/Sidebar.jsx [hunk 15]
```diff
@@ -107,8 +96,17 @@ function Sidebar() {
...
-                    {renderApplyFilterButton()}
-                </>
+                    {width === MOBILE_WIDTH &&
+                        (selectedCategory ||
+                            selectedProviders.length !== 0) && (
+                            <button
+                                className={styles['apply-filters-button']}
+                                onClick={applyFilters}
+                            >
+                                Применить
```
Comment (STDio96) [inline src/components/Sidebar/Sidebar.jsx:108]: can move that to `shared/messages.js`

File: src/redux/actions/data.js [hunk 16]
```diff
@@ -17,20 +17,12 @@ export const setWidth = width => ({
...
-    // TODO
-    /* getDataFromSeverFunction().then(({ data }) => {
-        // TODO: create an object with 4 arrays from the response
-        dispatch(setData(data))
-    }) */
+    getDataFromSeverFunction(token)
+        .then(data => data.json())
+        .then(body => {
+            dispatch(setData(body))
+            dispatch(setLoaded(true))
```
Comment (STDio96) [inline src/redux/actions/data.js]: I fixed that in my previous PRs, there's no need to manually set `isLoaded: true` here. It should be done via actions

File: src/components/AllGames/GamesContainer/GamesContainer.jsx [hunk 17]
```diff
@@ -31,7 +31,7 @@ function GamesContainer({ title, games, numberToDisplay = null, icon = null }) {
     return (
         <div className={styles['container']}>
             <div className={styles['header']}>
-                {icon ? <img src={icon} alt="icon" /> : <></>}
+                {icon && <img src={icon} alt="icon" />}
```
Comment (STDio96) [inline src/components/AllGames/GamesContainer/GamesContainer.jsx]: should be already fixed in master

File: src/components/Sidebar/Categories/Categories.jsx [hunk 18]
```diff
@@ -5,7 +5,7 @@ import { useSelector } from 'react-redux'
 import PropTypes from 'prop-types'
 
 Categories.propTypes = {
-    selectedCategory: PropTypes.array,
+    selectedCategory: PropTypes.string,
```
Comment (STDio96) [inline src/components/Sidebar/Categories/Categories.jsx]: already fixed

File: src/shared/slider/Slider.js [hunk 19]
```diff
@@ -6,25 +6,33 @@ import './slider.css'
 export const Slider = ({ children }) => {
     const responsive = {
         desktop: {
-            breakpoint: { max: 3000, min: 1024 },
+            breakpoint: { max: 3000, min: 761 },
```
Comment (STDio96) [inline src/shared/slider/Slider.js:12]: `max: 3000`? what about 8k monitors? :D

File: src/shared/modals/RegistrationModal.js [hunk 20]
```diff
@@ -25,13 +26,7 @@ const RegistrationModal = ({ setIsRegistrationModalVisible, setToken }) => {
             setRegistrationError('Пароли должны совпадать!')
```
Comment (STDio96) [inline src/shared/modals/RegistrationModal.js:26]: move to `shared/messages.js`

File: src/redux/actions/data.js (hunk 16 above)
Comment (STDio96) [reply src/redux/actions/data.js]: https://github.com/Virtual-Sports/Web/blob/8b2046b599f78d81146bb6235aa8374b6b129094/src/redux/reducers/data.js#L25

File: src/components/GamePage/GamePage.jsx (hunk 12 above)
Comment (VolodymyrButs) [reply src/components/GamePage/GamePage.jsx:23]: no) if we don't have game we run to dice

**Review by dkupchyk — APPROVED** [review]

**Pull request merged**